### PR TITLE
Base cluster with position/error, sensor ID

### PR DESCRIPTION
**Pull request by shahor02**
For alignment reason also the user managed bitmask and count fields are added

### CODE REVIEW

File: Detectors/Base/include/DetectorsBase/BaseCluster.h [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+  
+  // get sensor id
+  std::int16_t getSensorID()          const { return mSensorID; }
+  // get count field
+  std::int8_t getCount()              const { return mCount; }
+  // get bit field
+  std::uint8_t getBits()              const { return mBits; }
+  bool isBitSet(int bit)              const { return mBits & (0xff & (0x1<<bit)); }
+  // check special reserved bit to flag cluster misalignment
+  bool IsMisaligned()                 const  { return mBits & kMisalignMask;}
```
Comment (sawenzel) [inline Detectors/Base/include/DetectorsBase/BaseCluster.h]: `isMisaligned`?

Comment (shahor02) [inline Detectors/Base/include/DetectorsBase/BaseCluster.h]: Fixed/squashed

File: Detectors/Base/include/DetectorsBase/BaseCluster.h [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+  void setX(T x)                            { mPos.SetX(x); }
+  void setY(T y)                            { mPos.SetY(y); }
+  void setZ(T z)                            { mPos.SetZ(z); }
+  void setXYZ(T x, T y, T z)
+  {
+    setX(x);
+    setY(y);
+    setZ(z);
+  }
+  void setPos(Point3D<T> const &p)          { mPos = p; }
```
Comment (wiechula) [inline Detectors/Base/include/DetectorsBase/BaseCluster.h]: Why is here const after the type? Everywhere else it is before.

Comment (shahor02) [inline Detectors/Base/include/DetectorsBase/BaseCluster.h]: Just fixed for uniformity reason, though the meaning does not change

**Pull request merged**